modules/biscuit/align/main.nf

```diff
@@ -0,0 +1,47 @@
+process BISCUIT_ALIGN {
+    tag "$meta.id"
+    label 'process_high'
+
+    conda (params.enable_conda ? "bioconda::biscuit=1.0.2.20220113 bioconda::samtools=1.15" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mulled-v2-db16f1c237a26ea9245cf9924f858974ff321d6e:17fa66297f088a1bc7560b7b90dc273bf23f2d8c-0':
+        'quay.io/biocontainers/mulled-v2-db16f1c237a26ea9245cf9924f858974ff321d6e:17fa66297f088a1bc7560b7b90dc273bf23f2d8c-0' }"
+
+    input:
+    tuple val(meta), path(reads)
+    path index
+
+    output:
+    tuple val(meta), path("*.bam"), emit: bam
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def args2 = task.ext.args2 ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def read_group = meta.read_group ? "-R ${meta.read_group}" : ""
+    def biscuit_cpus = (int) Math.max(Math.floor(task.cpus/1.11),1)
```

Comment:
wow! is this cpu setting specific for biscuit?

Reply:
Yes... The goal is to assign the bulk of the available CPUs to the aligner. In cases where there's a lot of CPUs available though it's been my experience that it helps to pass samtools a few extra cores as well. This hopefully allows for that while avoiding specifying more cores than are actually available...

Looking at it again dividing by 1.11 seems a bit more obtuse than the equivalent 'task.cpus*0.9', so maybe I should change that to make it clearer.

Reply:
The alternatives I've seen are 

1. Pass `task.cpus` to both aligner and samtools (seems reckless when e.g. 20 cores are passed but then upto 40 are 'expected' by the tools, even if there's no way samtools would use 40 cores)
2. Pass `task.cpus` to just the aligner. This doesn't seem quite optimal as with a lot of cores, the aligner could 'pile up' reads in the pipe (producing alignments faster than samtools can consume them), degrading overall performance and possibly memory consumption.

Reply:
Happy to change if you think am alternative approach would be better

Reply:
I think `task.cpus * 0.9` would be fine here :+1: